javascript/javascript1/week4/homework.js

```diff
@@ -0,0 +1,120 @@
+        todo = voiceCom.slice(start, end);
+        let index = user.toDoList.indexOf(todo);
+        if (index > -1) {
+            user.toDoList.splice(index, 1);
+            return todo + "removed from your To Do List."
+        } else {
+            return "I couldn't find " + todo + "in your To Do List."
+        }
+
+    } else if (voiceCom.includes("What") && voiceCom.includes("todo")) {
+        if (user.toDoList.length === 0) {
+            return "Your To Do List is empty. Add some tasks.";
+        } else {
+            return user.toDoList;
+        }
+    } else if (voiceCom.includes("What") && voiceCom.includes("day") && voiceCom.includes("today")) {
+        const date = new Date();
+        let day = date.toLocaleDateString('en-GB', { day: 'numeric' });
+        let month = date.toLocaleDateString('en-GB', { month: 'long' });
+        let year = date.toLocaleDateString('en-GB', { year: 'numeric' })
+        today = day + " of " + month + " " + year;
+        return "Today is " + today;
+    } else if (voiceCom.includes("+") || voiceCom.includes("-") || voiceCom.includes("*") || voiceCom.includes("/")) {
+        let dig = /\d+/g;
+        let digits = voiceCom.match(dig);
+        let a = parseInt(digits[0]);
+        let b = parseInt(digits[1]);
+        if (voiceCom.includes("+")) {
+            return a + b;
+        } else if (voiceCom.includes("-")) {
+            return a - b;
+        } else if (voiceCom.includes("*")) {
+            return a * b;
+        } else if (voiceCom.includes("/")) {
+            return a / b;
+        }
+
+    } else if (voiceCom.includes("timer") && (voiceCom.includes("minute") || voiceCom.includes("second"))) {
+        let dig = /\d+/;
+        let time = voiceCom.match(dig);
+        let a = 0;
+        a = parseInt(time[0]);
+        let timReq = "";
+        if (voiceCom.includes("second")) {
+            a = a * 1000;
+            timeReq = a > 1000 ? "seconds" : "second";
+        } else if (voiceCom.includes("minute")) {
+            a = a * 60000;
+            timeReq = a > 60000 ? "minutes" : "minute";
+        }
+        setTimeout(function () {
+            console.log("Time's done");
+        }, a);
+
+        return "Timer set for " + time + " " + timeReq;
+
+    } else if (voiceCom.includes("What") && voiceCom.includes("time is")) {
+        const date = new Date();
+        let hour = date.getHours();
+        let minutes = date.getMinutes();
+        let time = "The time is " + hour + (hour > 1 ? " hours" : " hour") + " with " + minutes + (minutes > 1 ? " minutes" : " minute");
+        return time;
+    } else if (voiceCom.includes("What") && voiceCom.includes("your name")) {
+        return "Guess it! Hint: Is not Siri or Alexa."
+    } else {
+        return "I can't help you with that for the moment."
+    }
+
+
+
+}
+
+console.log(getReply("Hello my name is John"));
+console.log(getReply("Hello my name is Joaquin"));
+console.log(getReply("What is my name?"));
```

Comment:
Will this work for "what is my name?"
```suggestion
console.log(getReply("what is my name?"));
```

You could use toLowerCase() on your input command so that users can type in upper and lower case letters. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/toLowerCase